include/deal.II/particles/property_pool.h

```diff
@@ -115,14 +124,37 @@ namespace Particles
      */
     const unsigned int n_properties;
 
+    /**
+     * The currently allocated properties (whether assigned to
+     * a particle or available for assignment).
+     */
+    std::vector<double> properties;
+
     /**
      * A collection of handles that have been created by
-     * allocate_properties_array() and have not been destroyed by
-     * deallocate_properties_array().
+     * allocate_properties_array() and have been destroyed by
+     * deallocate_properties_array(). Since the memory is still
+     * allocated these handles can be reused for new particles
+     * to avoid memory allocation.
      */
-    std::set<Handle> currently_open_handles;
+    std::vector<Handle> currently_available_handles;
   };
 
+    /* ---------------------- inline and template functions ------------------ */
+
+  inline ArrayView<double>
+  PropertyPool::get_properties(const Handle handle)
+  {
+    const std::vector<double>::size_type data_index =
+      (handle != invalid_handle) ? handle * n_properties : 0;
+    Assert(data_index <= properties.size() - n_properties,
+           ExcMessage("Invalid property handle. This can happen if the "
+                      "handle was duplicated and then one copy was deallocated "
+                      "before trying to access the properties."));
+
+    return ArrayView<double>(properties.data() + data_index, n_properties);
```

Comment:
I think this might be easier to read:
```suggestion
    return ArrayView<double>(&properties[data_index], n_properties);
```

Reply:
It was me to request that code, so let me take the explanation here: The problem with dereferencing is that we would need another `if` here in case we do not have any properties and any memory allocated to avoid the invalid lookup - and in debug mode and with the debug enabled in the STL lib you actually trigger assertions.

Reply:
Ah, then by all means let's go with the code as is!